### PR TITLE
[TAS-950] 🚸 Display current Stripe Account

### PR DESCRIPTION
**Pull request by AuroraHuang22**
改動項目：

- Check if the login wallet has a Stripe account and display it under 'Use my account'; if not, prompt to create one
截圖 2024-04-18 晚上8 37 04
截圖 2024-04-18 晚上8 38 01

- Check the Stripe account info after entering the wallet.
截圖 2024-04-18 晚上8 40 40
截圖 2024-04-18 晚上8 40 49

- Add a save button for this section

### CODE REVIEW

**Comment by AuroraHuang22** [comment]
截圖 2024-04-23 上午9 57 22
[🔥 Hide Stripe Connect wallet address](https://github.com/likecoin/nft-book-press/pull/139/commits/762372b5f4e0166d31bd45714e4f01dac6655fd4)

File: pages/nft-book-store/collection/new.vue [hunk 1]
```diff
@@ -635,9 +697,44 @@ function addNotificationEmail () {
...
+  }
+  isStripeConnectLoading.value = true
+  stripeConnectStatusByWalletMap.value[inputValue] = { isReady: false }
+  try {
+    const { data: stripeConnectStatus, error } = await useFetch(
+      `${LIKE_CO_API}/likernft/book/user/connect/status?wallet=${inputValue}`
+    )
+    if (
+      error?.value &&
+      error?.value?.statusCode !== 404
```
Comment (williamchong) [inline pages/nft-book-store/collection/new.vue:721]: what if it's 404?

File: pages/nft-book-store/new.vue [hunk 2]
```diff
@@ -832,9 +822,44 @@ function addNotificationEmail () {
...
+  }
+  isStripeConnectLoading.value = true
+  stripeConnectStatusByWalletMap.value[inputValue] = { isReady: false }
+  try {
+    const { data: stripeConnectStatus, error } = await useFetch(
+      `${LIKE_CO_API}/likernft/book/user/connect/status?wallet=${inputValue}`
+    )
+    if (
+      error?.value &&
+      error?.value?.statusCode !== 404
```
Comment (williamchong) [inline pages/nft-book-store/new.vue:846]: what if it's 404?

File: pages/nft-book-store/collection/status/[collectionId].vue [hunk 3]
```diff
@@ -977,9 +1051,57 @@ function addNotificationEmail () {
...
+  }
+  isStripeConnectLoading.value = true
+  stripeConnectStatusByWalletMap.value[inputValue] = { isReady: false }
+  try {
+    const { data: stripeConnectStatus, error } = await useFetch(
+      `${LIKE_CO_API}/likernft/book/user/connect/status?wallet=${inputValue}`
+    )
+    if (
+      error?.value &&
+      error?.value?.statusCode !== 404
```
Comment (williamchong) [inline pages/nft-book-store/collection/status/[collectionId].vue:1071]: What if it's 404? Also this is the third time we have this code snippet, maybe change it to component or composable

**Review by williamchong — CHANGES_REQUESTED** [review]
Recommend to refactor the stripe connect input/status ui and its logic

File: stores/stripe.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import { LIKE_CO_API } from '~/constant'
+
+export const useStripeStore = defineStore('stripe-connect', () => {
+  const stripeConnectStatusWalletMap = ref({} as Record<string, any>)
+
+  async function fetchStripeConnectStatus (wallet: string) {
+    stripeConnectStatusWalletMap.value[wallet] = { isReady: false }
+    const connectStatusData = await useFetch(
+        `${LIKE_CO_API}/likernft/book/user/connect/status?wallet=${wallet}`
+    )
```
Comment (williamchong) [inline stores/stripe.ts:11]: Should throw if `stripeConnectStatusWalletMap.error` is trueful

**Review by williamchong — CHANGES_REQUESTED** [review]

File: stores/stripe.ts (hunk 4 above)
Comment (AuroraHuang22) [reply stores/stripe.ts]: [🎨 Improve error handling](https://github.com/likecoin/nft-book-press/pull/139/commits/dc3b1af80b3dfcbd2f31238d39c2d9c31dc5df47)

**Review by williamchong — CHANGES_REQUESTED** [review]
Prefer `v-model` in the new component instead of manually naming the events

File: pages/nft-book-store/collection/new.vue [hunk 5]
```diff
@@ -186,15 +186,13 @@
       </UCard>
 
       <StripeConnectCard
-        :is-stripe-connect-checked="isStripeConnectChecked"
+        v-model="isStripeConnectChecked"
```
Comment (williamchong) [inline pages/nft-book-store/collection/new.vue:189]: name this as `v-model:isStripeConnectChecked` also?

**Review by williamchong — CHANGES_REQUESTED** [review]

File: pages/nft-book-store/collection/new.vue (hunk 5 above)
Comment (AuroraHuang22) [reply pages/nft-book-store/collection/new.vue]: [🎨 Improve v-model bindings for clarity](https://github.com/likecoin/nft-book-press/pull/139/commits/dc3caa7654325d5af46e597b78898d21dc2e06e6)

**Comment by AuroraHuang22** [comment]
defineModel is enabled by default starting from version 3.10.0
https://github.com/nuxt/nuxt/pull/25306

File: components/StripeConnectCard.vue [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+            :disabled="!!props.shouldDisableSetting"
+            class="w-[50%]"
+          >
+            <template #label>
+              <span>Use another Stripe Express account</span>
+              <div class="flex flex-col my-[10px]">
+                <UInput
+                  :value="inputWallet"
+                  :color="stripeConnectInputError ? 'rose' : 'white'"
+                  class="font-mon w-full"
```
Comment (nwingt) [inline components/StripeConnectCard.vue:68]:
```suggestion
                  class="font-mono w-full"
```

File: components/StripeConnectCard.vue [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+    }"
+  >
+    <template #header>
+      <h3 class="font-bold font-mono">
+        Connect to a Stripe Account
+      </h3>
+      <UToggle
+        v-model="isStripeConnectChecked"
+        name="stripe"
+        label="Use a Stripe Connect account for receiving all payment"
```
Comment (nwingt) [inline components/StripeConnectCard.vue:18]:
```suggestion
        label="Use a Stripe Connect account for receiving payments"
```

File: components/StripeConnectCard.vue [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+                  v-if="stripeConnectInputError"
+                  class="text-red-700 text-[10px]"
+                >{{ stripeConnectInputError }}</span>
+              </div>
+              <div v-if="isStripeConnectLoading" class="text-center">
+                Loading ...
+              </div>
+              <div
+                v-else-if="
+                  props.stripeConnectStatusWalletMap[inputWallet]
```
Comment (nwingt) [inline components/StripeConnectCard.vue:82]: this can be computed

File: components/StripeConnectCard.vue [hunk 9]
```diff
@@ -0,0 +1,218 @@
...
+                    v-if="
+                      props.stripeConnectStatusWalletMap[inputWallet]
+                        ?.email
+                    "
+                  >{{
+                    ` Email: ${props.stripeConnectStatusWalletMap[inputWallet]?.email}`
+                  }}</span>
+                </div>
+                <div v-else>
+                  ❌ No stripe account connected to this wallet yet.<br>
```
Comment (nwingt) [inline components/StripeConnectCard.vue:100]: Try to make use of Nuxt UI component like `<UAlert/>`

File: components/StripeConnectCard.vue [hunk 10]
```diff
@@ -0,0 +1,218 @@
...
+                  <span> ✅ Has Stripe Account</span><br>
+                  <span
+                    v-if="
+                      props.stripeConnectStatusWalletMap[inputWallet]
+                        ?.email
+                    "
+                  >{{
+                    ` Email: ${props.stripeConnectStatusWalletMap[inputWallet]?.email}`
+                  }}</span>
+                </div>
```
Comment (nwingt) [inline components/StripeConnectCard.vue:98]: Try to make use of Nuxt UI component like `<UAlert/>`

File: components/StripeConnectCard.vue [hunk 11]
```diff
@@ -0,0 +1,218 @@
...
+                  @input="onStripeConnectWalletInput"
+                />
+                <span
+                  v-if="stripeConnectInputError"
+                  class="text-red-700 text-[10px]"
+                >{{ stripeConnectInputError }}</span>
+              </div>
+              <div v-if="isStripeConnectLoading" class="text-center">
+                Loading ...
+              </div>
```
Comment (nwingt) [inline components/StripeConnectCard.vue:79]: Try to make use of Nuxt UI component like `<UProgress animation="carousel" />`

File: components/StripeConnectCard.vue (hunk 6 above)
Comment (AuroraHuang22) [reply components/StripeConnectCard.vue]: [💄Adjust style & wording](https://github.com/likecoin/nft-book-press/pull/139/commits/a6b3d1fc776e8e2e295dfe49b9e4e79b3f745315)

File: components/StripeConnectCard.vue (hunk 7 above)
Comment (AuroraHuang22) [reply components/StripeConnectCard.vue]: [💄Adjust style & wording](https://github.com/likecoin/nft-book-press/pull/139/commits/a6b3d1fc776e8e2e295dfe49b9e4e79b3f745315)

File: components/StripeConnectCard.vue (hunk 8 above)
Comment (AuroraHuang22) [reply components/StripeConnectCard.vue]: [💄Adjust style & wording](https://github.com/likecoin/nft-book-press/pull/139/commits/a6b3d1fc776e8e2e295dfe49b9e4e79b3f745315)

File: components/StripeConnectCard.vue (hunk 9 above)
Comment (AuroraHuang22) [reply components/StripeConnectCard.vue]: [💄Adjust style & wording](https://github.com/likecoin/nft-book-press/pull/139/commits/a6b3d1fc776e8e2e295dfe49b9e4e79b3f745315)

File: components/StripeConnectCard.vue (hunk 10 above)
Comment (AuroraHuang22) [reply components/StripeConnectCard.vue]: [💄Adjust style & wording](https://github.com/likecoin/nft-book-press/pull/139/commits/a6b3d1fc776e8e2e295dfe49b9e4e79b3f745315)

File: components/StripeConnectCard.vue (hunk 11 above)
Comment (AuroraHuang22) [reply components/StripeConnectCard.vue]: [💄Adjust style & wording](https://github.com/likecoin/nft-book-press/pull/139/commits/a6b3d1fc776e8e2e295dfe49b9e4e79b3f745315)

File: components/StripeConnectCard.vue (hunk 9 above)
Comment (AuroraHuang22) [reply components/StripeConnectCard.vue]: [💄Update radio group style](https://github.com/likecoin/nft-book-press/pull/139/commits/d591169181cccc1b7d476837e23aadce8886512a)

File: components/StripeConnectCard.vue (hunk 10 above)
Comment (AuroraHuang22) [reply components/StripeConnectCard.vue]: [💄Update radio group style](https://github.com/likecoin/nft-book-press/pull/139/commits/d591169181cccc1b7d476837e23aadce8886512a)

File: stores/stripe.ts [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+  const stripeConnectStatusWalletMap = ref({} as Record<string, any>)
+
+  async function fetchStripeConnectStatus (wallet: string) {
+    stripeConnectStatusWalletMap.value[wallet] = { isReady: false }
+    const { data, error } = await useFetch(
+        `${LIKE_CO_API}/likernft/book/user/connect/status?wallet=${wallet}`
+    )
+    stripeConnectStatusWalletMap.value[wallet] = (data?.value as any) || {}
+    if (error.value) {
+      throw new Error(error.value?.data.toString())
```
[the remark this answers is not in the document]
Comment (AuroraHuang22) [reply stores/stripe.ts]: [🚸 Refine error handling in fetchStripeConnectStatus](https://github.com/likecoin/nft-book-press/pull/139/commits/e97dd7786fea1319b21ba3ef299f89f0628ddbac)

File: components/StripeConnectCard.vue (hunk 9 above)
Comment (AuroraHuang22) [reply components/StripeConnectCard.vue]: 截圖 2024-04-29 下午6 54 08

File: stores/stripe.ts [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+    stripeConnectStatusWalletMap.value[wallet] = { isReady: false }
+    const { data, error } = await useFetch(
+        `${LIKE_CO_API}/likernft/book/user/connect/status?wallet=${wallet}`
+    )
+    stripeConnectStatusWalletMap.value[wallet] = (data?.value as any) || {}
+    if (error.value) {
+      if (error.value.statusCode !== 404) {
+        throw new Error(error.value?.data.toString())
+      // eslint-disable-next-line no-console
+      } console.error('STRIPE_CONNECT_INFO_NOT_FOUND')
```
Comment (williamchong) [inline stores/stripe.ts:17]:
```suggestion
      }
      // eslint-disable-next-line no-console
      console.error('STRIPE_CONNECT_INFO_NOT_FOUND')
```

Comment (AuroraHuang22) [reply stores/stripe.ts]: [🎨 Improve formatting](https://github.com/likecoin/nft-book-press/pull/139/commits/3f18df0210c78be033efc3ff2ad06db62e2c2d0b)

**Review by williamchong — APPROVED** [review]

**Review by nwingt — APPROVED** [review]

**Pull request merged**